odoo/tests/common.py

```diff
@@ -1766,7 +1779,8 @@ def authenticate(self, user, password):
             # than this transaction.
             self.cr.flush()
             self.cr.clear()
-            uid = self.registry['res.users'].authenticate(session.db, user, password, {'interactive': False})
+            with patch('odoo.addons.base.models.res_users.Users._check_credentials'):
```

Comment:
I don't understand the purpose of this one.  It patches `Users._check_credentials` with a `MagicMock`, and the latter simply returns a `Mock` when it is called...

Reply:
yes, _check_credentials works by side_effect, raising if the credential does not match. Patching it is enough to disable the check.

```python
if not valid:
            raise AccessDenied()
```
But the comparaison is slow since it is using 350000 rounds, pbkdf2_hmac represents almost of the authenticate time. This is juste a small optional performance increase allowing to win ~300ms per httpcase.